tests/integration/test_stepfunctions.py

```diff
@@ -570,6 +581,36 @@ def assert_success():
         client1.delete_state_machine(stateMachineArn=child_machine_arn)
 
 
+@pytest.mark.aws_validated
+def test_default_logging_configuration(
+    iam_client,
+    stepfunctions_client,
+):
+    role_name = f"role_name-{short_uid()}"
+    role_arn = iam_client.create_role(
+        RoleName=role_name,
+        AssumeRolePolicyDocument=json.dumps(STS_ROLE_POLICY_DOC),
+    )["Role"]["Arn"]
+
+    definition = clone(TEST_STATE_MACHINE)
+    definition = json.dumps(definition)
+
+    sm_name = f"sts-logging-{short_uid()}"
+    result = stepfunctions_client.create_state_machine(
+        name=sm_name, definition=definition, roleArn=role_arn
+    )
+
+    assert result["ResponseMetadata"]["HTTPStatusCode"] == 200
+    result = stepfunctions_client.describe_state_machine(stateMachineArn=result["stateMachineArn"])
+    assert result["ResponseMetadata"]["HTTPStatusCode"] == 200
+    assert result["loggingConfiguration"]
+    assert result["loggingConfiguration"] == {"level": "OFF", "includeExecutionData": False}
+
+    # clean up
+    stepfunctions_client.delete_state_machine(stateMachineArn=result["stateMachineArn"])
+    iam_client.delete_role(RoleName=role_name)
```

Comment:
unsafe cleanup

This obv. isn't super critical when running against LocalStack but a test against AWS might leak resources here.

Either fixtures or other safer cleanup strategies should be used instead. A finally block would be ok too but would again need to wrap the individual cleanup statements in a safe way so they don't interfere with each other.